lib/main.dart

```diff
@@ -1,42 +1,74 @@
+import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:game_lib_app/cubit/games_cubits.dart';
+import 'package:game_lib_app/blocs_and_cubits/auth/auth_bloc.dart';
+import 'package:game_lib_app/blocs_and_cubits/drawer_cubit.dart';
+import 'package:game_lib_app/blocs_and_cubits/user_cubit.dart';
 
 import 'package:game_lib_app/locale_string.dart';
+import 'package:game_lib_app/repositories/fb_auth_repository.dart';
+import 'package:game_lib_app/repositories/firestore_repository.dart';
 import 'package:game_lib_app/services/network_service.dart';
 import 'package:game_lib_app/views/main_view/main_view.dart';
 import 'package:get/get.dart';
+import 'firebase_options.dart';
 
-void main() {
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
   runApp(const MyApp());
 }
 
 class MyApp extends StatelessWidget {
   const MyApp({Key? key}) : super(key: key);
   @override
   Widget build(BuildContext context) {
-    return MultiBlocProvider(
+    return MultiRepositoryProvider(
       providers: [
-        BlocProvider<FavGamesCubit>(
-          create: (context) => FavGamesCubit(),
-        ),
-        BlocProvider<PlayedGamesCubit>(
-          create: (context) => PlayedGamesCubit(),
-        ),
-        BlocProvider<WishlistGamesCubit>(
-          create: (context) => WishlistGamesCubit(),
+        RepositoryProvider(
+          create: (context) => AuthRepository(
+              firebaseAuth: FirebaseAuth.instance,
+              firebaseFirestore: FirebaseFirestore.instance),
         ),
+        RepositoryProvider(
+          create: (context) => FirestoreRepository(
+              firebaseFirestore: FirebaseFirestore.instance),
+        )
       ],
-      child: GetMaterialApp(
-        title: 'Game Library',
-        theme: ThemeData(
-          scaffoldBackgroundColor: Colors.grey[900],
-          primarySwatch: Colors.purple,
-          primaryColor: Colors.purple[600],
+      child: MultiBlocProvider(
+        providers: [
+          BlocProvider<AuthBloc>(
+            create: (context) =>
+                AuthBloc(authRepository: context.read<AuthRepository>()),
+          ),
+          BlocProvider<UserCubit>(
+            create: (context) => UserCubit(
+                firestoreRepository: FirestoreRepository(
+                    firebaseFirestore: FirebaseFirestore.instance)),
+          ),
+        ],
+        child: GetMaterialApp(
+          title: 'Game Library',
+          theme: ThemeData(
+            scaffoldBackgroundColor: Colors.grey[900],
+            primarySwatch: Colors.purple,
+            primaryColor: Colors.purple[600],
+          ),
+          locale: const Locale('en', 'US'),
+          translations: LocaleString(),
+          home: BlocProvider<DrawerCubit>(
+            create: (context) => DrawerCubit(),
```

Comment:
Why this BlocProvider isn't added with another providers? And also I think that you should add signup/signin providers above the material app, to the wrapper that I've wrote you before. I think that it will be cleaner.

Reply:
Because it is used only under this scope, I thought that I should wrap my tree with bloc providers as low as possible. The difference is that when it is here it is not visible in other navigation routes. When it is above Material app it would be possible to access it from other routes and e.g. alert dialogs

Reply:
Okay, yeah maybe you're right. If you know in advance that you will be using this cubit only in one scope your solution is good :D